core/vm/gas_table.go

```diff
@@ -55,45 +55,62 @@ func memoryGasCost(mem *Memory, newMemSize uint64) (uint64, error) {
 	return 0, nil
 }
 
+func memoryGasCostAndSize(stack *Stack, mem *Memory, sizeFunc memorySizeFunc) (uint64, uint64, error) {
```

Comment:
```suggestion
func memoryGasCostAndSize(stack *Stack, mem *Memory, sizeFunc memorySizeFunc) (gas, memorySize uint64, err error) {
```

You can name results (although it is still confusing for me that now these variables are defined from the beginning of the function).

Reply:
Thanks, I forgot about that, but not sure if I like this variant more, and whether it aligns well with the usual geth style.
```go
func memoryGasCostAndSize(stack *Stack, mem *Memory, sizeFunc memorySizeFunc) (gas uint64, memorySize uint64, err error) {
	memSize, overflow := sizeFunc(stack)
	if overflow {
		return 0, 0, ErrGasUintOverflow
	}
	// memory is expanded in words of 32 bytes. Gas is also calculated in words.
	memorySize, overflow = math.SafeMul(toWordSize(memSize), 32)
	if overflow {
		return 0, 0, ErrGasUintOverflow
	}
	gas, err = memoryGasCost(mem, memorySize)
	if err != nil {
		return 0, 0, err
	}
	return
}
```